cmake/AwsPrebuildDependency.cmake

```diff
@@ -75,3 +92,30 @@ function(aws_prebuild_dependency)
         DESTINATION ${CMAKE_INSTALL_PREFIX}
     )
 endfunction()
+
+# Get list of optional variables that may affect build process.
+function(aws_get_variables_for_prebuild_dependency AWS_CMAKE_PREBUILD_ARGS)
+    set(variables "")
+    set(variablesToIgnore CMAKE_INSTALL_PREFIX)
+
+    get_cmake_property(vars CACHE_VARIABLES)
+    foreach(var ${vars})
+        message("= Checking ${var}")
+        if (var MATCHES "^(CMAKE_)?ANDROID_"
+                OR var STREQUAL "CMAKE_TOOLCHAIN_FILE"
+                OR var STREQUAL "CMAKE_SYSTEM_NAME"
```

Comment:
in a normal build, is CMAKE_SYSTEM_NAME in the cache? and therefore always passed to the prebuild?

I see that this affects [CMAKE_CROSSCOMPILING](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html)
> This variable will be set to true by CMake if the CMAKE_SYSTEM_NAME variable has been set manually (i.e. in a toolchain file or as a cache entry from the cmake command line). In most cases, manually setting CMAKE_SYSTEM_NAME will only be done when cross compiling since, if not manually set, it will be given the same value as CMAKE_HOST_SYSTEM_NAME, which is correct for the non-cross-compiling case. In the event that CMAKE_SYSTEM_NAME is manually set to the same value as CMAKE_HOST_SYSTEM_NAME, then CMAKE_CROSSCOMPILING will still be set to true.

Is that OK?

Are there other weird side-effects to worry about, due to manually passing stuff in the prebuild, that got into the main build's cache automatically?

Reply:
I checked on Linux and Mac, `CMAKE_SYSTEM_NAME` is not set.
However, to play it safe, I'll add a `CMAKE_CROSSCOMPILING` check along with `CMAKE_SYSTEM_NAME`.

Reply:
OK, I put all vars that make sense only in cross-compiling mode to a separate `if` block checking additionally for `CMAKE_CROSSCOMPILING` var.